File: Frontend/src/app/models/admin/menu-items.ts
import { Functionalities } from "./functionalities";
import { MenuHeader } from "./menu-headers";

export class MenuItems {
    id?: number;
    header_id?: number;
    menuHeader?: MenuHeader;
    menuHeaders?: MenuHeader[];
    item_label?: string;
    item_icon?: string;
    item_to?: string;
    item_order?: number;
    item_badge?: string;
    maker_id?: number;
    maker_email?: string;
    status?: boolean;
    created_date?: string;
    updated_date?: string;
    delete?: boolean;
    permissions?: Functionalities[]
}

File: Backend/TaxManagementSystem/src/main/java/com/tms/Admin/Entity/MenuItems.java
package com.tms.Admin.Entity;

import java.util.List;

import com.tms.Common.Entity.Functionalities;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Data;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class MenuItems {
	private Long id;
	private Long header_id;
	private String item_label;
	private String item_icon;
	private String item_to;
	private String item_badge;
	private Long maker_id;
	private String maker_email;
	private MenuHeader menuHeader;
	private List<MenuHeader> menuHeaders;
	private String created_date;
	private String updated_date;
	private boolean status;
	private int item_order;
	private boolean delete;
	private List<Functionalities> permissions;

}
